### PR TITLE
[perf only] Test perf regression from copying all parameters

### PR DESCRIPTION
**Pull request by JakobDegen**
cc @bjorn3 can you check that this is the right change, and start a perf run if so?

cc #71117

### CODE REVIEW

**Comment by JakobDegen** [comment]
Oh, can I

r? ghost

or something?

**Comment by JakobDegen** [comment]
Does it not work outside of the original message, or do I have to do it?

r? @ghost

**Comment by JakobDegen** [comment]
Oh apparently I have perms for this now

@bors try @rust-timer queue

File: compiler/rustc_codegen_ssa/src/mir/block.rs [hunk 1]
```diff
@@ -850,6 +850,7 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             // by-ref, so make a local copy of non-immediate constants.
             match (arg, op.val) {
                 (&mir::Operand::Copy(_), Ref(_, None, _))
+                | (&mir::Operand::Move(_), Ref(_, None, _))
```
Comment (bjorn3) [inline compiler/rustc_codegen_ssa/src/mir/block.rs:853]: I'm pretty sure this is correct.

**Comment by JakobDegen** [comment]
The regressions are actually not as bad as I expected them to be. Dest prop should be able to recover most/all of this and then some, the only problem is that I'm not sure if we'll be able to run it on debug builds. I think I may try fixing dest prop under the assumption that this is what we do (or that we don't have to do it, whichever) and seeing what the perf results of that net change look like

**Pull request closed without merging**